en/guide/xml_schema.md

```diff
@@ -78,26 +84,36 @@ As a concrete example, the definition of the [LANDING_TARGET_TYPE](../messages/c
     </entry>
 ```
 
-The main `enum` tags/fields are:
-* `name`: The name of the enum (mandatory). This is a string of capitalized, underscore-separated words.
-* `description` (optional): A string describing the purpose of the enum
-* `entry` (optional): An entry (zero or more entries can be specified for each enum)
-* [deprecated](#deprecated) (optional): A tag indicating that the enum is deprecated.
-
 
 #### entry {#entry}
 
-The enum `entry` tags/fields are:
+The "normal" enum `entry` tags/fields are:
 * `name`: The name of the enum value (mandatory). This is a string of capitalized, underscore-separated words.
 * `value` (optional): The *value* for the entry (a number).
 * `description` (optional): A description of the entry.
-* [param](#param) (optional): A param value used in a [MAV_CMD](../messages/common.md#MAV_CMD) enum entry.
 * [deprecated](#deprecated) / [wip](#wip) (optional): A tag indicating that the enum is deprecated or "work in progress". 
 
+[MAV_CMD](../messages/common.md#MAV_CMD) entries may additionally define these tags/fields:
+* [param](#param) (optional): A param value.
+* `hasLocation` (optional): A boolean (default true) indicating the entry has lat/lon/alt location information in its param values. 
+* `isDestination` (optional):  A boolean (default true) indicating that the entry is a destination.
+
 #### param {#param}
 
 The `<param>` tag is used in the [MAV_CMD](../messages/common.md#MAV_CMD) enum as part of defining mission commands.
-Params include an attribute `index` (with value from 1 to 7) and contain a parameter description string.
+Params must include an attribute `index` (with value from 1 to 7) and contain a parameter description string.
+
+A `param` may also include the following optional attributes (which may be used by a GUI for parameter display and editing):
+- `label` - Display name to represent the parameter in a GCS or other UI
+- `units` - SI units for the value.
+- `enum` - Enum containing possible values for the parameter (if applicable).
```

Comment:
@amilcarlucas Normally enum values are named after the enum name. So, MAV_BEER enum, has items MAV_BEER_GOOD, and MAV_BEER_INDIFFERENT. But what is the pattern for enums to use in params? They can't all be MAV_CMD_WHATEVER?

Thinking of [MAV_CMD_NAV_LAND](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_LAND), param 2:

> Precision land mode. (0 = normal landing, 1 = opportunistic precision landing, 2 = required precsion landing)

Perhaps we simply ignore the MAV_CMD prefix and use the suffix. ie enum LAND_MODE, with values LAND_MODE_NORMAL, LAND_MODE_OPPORTUNISTIC_PRECISION_LANDING ...

Thoughts?

Reply:
That might unfortunately lead to name collisions in the generated C code :(

Reply:
@amilcarlucas I would assume that the generator should automatically check for any name collisions of any kind during validation/generation - right?

If not, then it should certainly (at a minimum) provide a warning.

If not, what would you suggest?

Reply:
Yes, it should check for name collisions, but AFAIK it does not do that at the moment.

Reply:
@amilcarlucas 
Following testing I can confirm that the generator does check for enum name collisions, so we can use any naming scheme we like. We still need to check that each enum value is unique though - because otherwise generated pymavlink dialects break.

Thinking about this some more, there is absolutely no reason that enums associated with MAV_CMD params should be named according to the parent enum (as I stated/inferred above). So we should just name the enum for each purpose sensibly for each case.

FYI only, what is tested for duplicates/not tested is covered here: https://github.com/ArduPilot/pymavlink/pull/233